dev-guide/cluster-fleet-evaluation.md

```diff
@@ -0,0 +1,162 @@
+contact with customers is made), but are free to drop options if they feel using
+that method may not reach the proper audience or may be too expensive.
+
+**Note**: Before the team goes down this path it will be useful to discuss the
+outcome and expectations with their respective OpenShift staff engineer, product
+manager and PLM contact. PLM contacts are listed under *column S* in the
+[OCP Team Member Tracking sheet](https://docs.google.com/spreadsheets/d/1M4C41fX2J1nBXhqPdtwd8UP4RAx98NA4ByIUv-0Z0Ds/edit#gid=1382138347).
+
+## Reference
+
+### Interacting with telemetry
+
+#### Pre-requisites
+
+To access the telemetry data please follow
+[these instructions](https://help.datahub.redhat.com/docs/interacting-with-telemetry-data).
+Once you have access, you will be able to perform queries against the data
+collected from customer clusters.
+
+#### Running queries
+
+You can query the telemetry data either using the
+[Thanos](https://telemeter-lts.datahub.redhat.com/graph) or
+[Grafana](https://telemeter-lts-dashboards.datahub.redhat.com) interfaces. For
+example, if you have introduced a new _ConditionStatus/Reason_ called _FooBar_,
+it can be queried using:
+
+```sql
+count by (_id, ebs_account) (
+  cluster_operator_conditions{name="component-name",condition="ClusterFleetEvaluation",reason="FooBar"}
+  * on (_id) group_left (ebs_account)
+  topk by (_id) (1, id_version_ebs_account_internal:cluster_subscribed)
+)
+```
+
+This will return a list of clusters and the associated customers. The above
+query is just an example and the team is free to create their own custom
+queries.
+
+### Writing a KCS document
+
+[KCS](https://source.redhat.com/groups/public/cee-kcs-program/cee_kcs_program_wiki/kcs_solutions_content_standard_v20)
+docs are split into solutions and articles.
+[Solutions](https://access.redhat.com/node/add/kcs-solution) explain breakages
+and how to fix them and are transactional in nature. The scenario presented here
+would typically result in a KCS solution.
+[Articles](https://access.redhat.com/node/add/kcs-article) are targeted towards
+highlighting features / explanations that are outside the scope of the OpenShift
+documentation. If you are new to writing a KCS doc, you can reach out to 
+[Albert Myles](amyles@redhat.comto) who can connect you with a KCS coach.
+
+### Alerting customers
+
+Once the component team has collected the telemetry data and the list of
+affected customers, they can alert them using any combination of the methods
+listed below. This should only be done if the impact of the change being
+introduced is very low. For help in facilitating use of any of these methods you
+can reach out to your PLM contact.The way to choose method(s) are subjective to
+the scope and impact of the change and should ideally be done in collaboration
+with your respective OpenShift staff engineer, product manager and PLM contact.
+
+In order of preference, in what teams should choose:
```

Comment:
not mentioned in this doc yet would be backporting a ClusterOperator `Upgradeable=False` guard to 4.(y-1) to keep folks from accidentally updating to 4.y until they'd become compatible.

Reply:
I am not sure that there is an intent to do that this as part of this guide. This is more focused on the evaluating the fleet.

Reply:
I think @wking's suggestion is that "setting upgradeable=false (with an appropriate reason)" is an "Alerting customers" mechanism.

In other words, a team might:

1) backport a change to survey the fleet
2) conclude that enough of the fleet is impacted that they need to notify people to make changes on their cluster prior to upgrading
3) use an upgradeable=false setting to notify impacted clusters

it's definitely not a preferred approach, but in situations where the percentage of impacted clusters is small (so it's not worth us holding the change until most clusters are compliant, because most already are), and the impact of upgrading w/o addressing the situation is high, it might be the right choice.

The primary motivation of this doc in the first place was we had cases where teams delivered changes that either blocked upgrades or put scary alerts in front of admins, without realizing just how many clusters were going to be impacted.  The goal was never to ensure that *no* clusters were impacted, but how much care/feeding we do is directly related to how many clusters will be affected (and what the effect is).  So if your survey results come back as "20% are impacted" then no, setting upgradeable=false is probably not the right answer.  If it's 2%?  it might be.

Reply:
> I think @wking's suggestion is that "setting upgradeable=false (with an appropriate reason)" is an "Alerting customers" mechanism.

That makes sense worded that way. I will add a paragraph about this without going into percentages.

Reply:
Added a paragraph at the end.